### DISCUSSION TITLE
Making a part of a vscode snippet library opt in

### DISCUSSION THREAD

**Discussion by blyedev** [L3MON4D3/LuaSnip#1241] [Q&A]
`rafamadriz/friendly-snippets` currently attributes framework specific snippets to `framework` filetype so for example angulars snippets are all bundled into an angular filetype. I am working on a pr to divide them up as seeing both html and ts snippets in each of those filetypes is significant noise especially if you enable multiple frameworks. The issue however is if they are bound to html and ts filetypes that disables their opt in character. That poses the question, aside from dividing this up into multiple repos is there any way to make the framework snippets opt in while preserving the filetypes of each snippet subset?

Right now my proposed solution is to just make a filetype like angular-html and make the user enable it for each ft but that makes the user write ~4 lines of configuration for each framework to enable it and that's pretty verbose despite the better scoping of snippets.

**Comment by L3MON4D3** [comment]
Hi :)
IMO the approach you describe at the end, using filetypes like `angular-html` to selectively enable project-specific snippets is exactly the right thing to do since then the simple `ls.filetype_extend("html", {"angular-html"})` is enough to enable them (or one could use `ft_func` to enable these snippets on a per-buffer basis, but that needs some more scaffolding)
